lib/mockito.dart

```diff
@@ -48,6 +53,119 @@ class Mock {
   String toString() => _givenName != null ? _givenName : runtimeType.toString();
 }
 
+// Return a new [Invocation], reconstituted from [invocation], [_typedArgs],
+// and [_typedNamedArgs].
+Invocation _reconstituteInvocation(Invocation invocation) {
+  var newInvocation = new FakeInvocation(invocation);
+  _typedArgs.clear();
+  _typedNamedArgs.clear();
+  return newInvocation;
+}
+
+/// An Invocation class with a functional constructor.
+class FakeInvocation extends Invocation {
+  final Symbol memberName;
```

Comment:
Is it valid for a field to override a getter? I thought that wasn't allowed anymore. Or was it the other way around?